--- SetCookie.php	2015-05-11 18:17:05.000000000 +0200
+++ SetCookie2.php	2015-06-03 14:36:08.000000000 +0200
@@ -163,7 +163,7 @@
             throw new Exception\InvalidArgumentException('Invalid header line for Set-Cookie string: "' . $name . '"');
         }
 
-        $multipleHeaders = preg_split('#(?<!Sun|Mon|Tue|Wed|Thu|Fri|Sat),\s*#', $value);
+        $multipleHeaders = preg_split('#(?<!Sun|Sunday|Mon|Monday|Tue|Tuesday|Wed|Wednesday|Thu|Thursday|Fri|Friday|Sat|Saturday),\s*#', $value);
 
         if (count($multipleHeaders) <= 1) {
             return $setCookieProcessor(array_pop($multipleHeaders));
